model/src/pyrenew/metaclass.py

```diff
@@ -721,6 +720,10 @@ def sample(self, **kwargs) -> tuple:
             for t, uv in zip(self.transforms, untransformed_values)
         )
 
+        numpyro.deterministic(self.name, transformed_values[0].value)
```

Comment:
1. Make optional (following the same `if record` pattern used in `DeterministicVariable`
2. Iterate over tuple entries in some sensible way and name the `numpyro.deterministic` sites accordingly?

Reply:
made these changes in https://github.com/CDCgov/multisignal-epi-inference/pull/387/commits/4b48a379b2b938f307e04a921c97c50bc29c1dde
Taking suggestions for better naming inside the loop (if any)!